# [1] colab 시작하기

- 분석환경 colab 이해
- 코드셀과 텍스트셀의 차이 알기



In [1]:
word = 'Hello Python'

Colab Notebook 안에서는 코드 셀과 텍스트 셀을 통해 코드를 작성하고, 설명할 수 있습니다

In [2]:
print(word)

Hello Python


## 셀 실행 방식의 특징

대화형 환경이기 때문에!

In [3]:
# x = 10
# x

10

## 실습1

In [4]:
users = [
    {"user_id":1,"age":25,"country":"KR","is_paid":False},
    {"user_id":2,"age":32,"country":"KR","is_paid":True},
    {"user_id":3,"age":29,"country":"US","is_paid":False},
    {"user_id":4,"age":41,"country":"US","is_paid":True},
    {"user_id":5,"age":36,"country":"KR","is_paid":True},
]

- 딕셔너리: key-value의 한 쌍
  - 순서 x, 변경o, 중복x(key에 대한 중복)
- 리스트 : 값이 연속된 공간에 순서가 있게 저장되어 있는 자료형
  - 순서o, 변경o, 중복o


In [5]:
type(users)

list

In [7]:
users[0]

{'user_id': 1, 'age': 25, 'country': 'KR', 'is_paid': False}

In [8]:
users[0]['age']

25


1. 유료 사용자(is_paid == True)의 나이만 출력해 보세요.

In [11]:
for user in users:
  if user['is_paid'] ==  True:
    print(user['age'])

32
41
36


2. 나이가 30세 이상인 사용자의 수를 출력해 보세요.

In [13]:
cnt = 0
for user in users:
  if user['age'] >= 30:
    cnt += 1
cnt

3

3. 전체 사용자의 평균 나이를 계산해 보세요.

In [14]:
sum_age = 0
for user in users:
  sum_age += user['age']

avg_age = sum_age/len(users)
avg_age


32.6

4. 유료 사용자만의 평균 나이를 계산해 보세요.

In [18]:
paid_age = 0
paid_cnt = 0
for user in users:
  if user['is_paid'] == True:
    paid_age += user['age']
    paid_cnt += 1

avg_age2 = paid_age/paid_cnt
print(round(avg_age2,1))

36.3


(1) 전체 사용자 수  

total_users =len(users)
print(total_users)

(2) 유료 사용자 수  

sum_age = 0
for user in users:
  sum_age += user['age']

avg_age = sum_age/len(users)
avg_age  

(3) 유료 사용자 비중

paid_age = 0
paid_cnt = 0
for user in users:
  if user['is_paid'] == True:
    paid_age += user['age']
    paid_cnt += 1

avg_age2 = paid_age/paid_cnt
print(round(avg_age2,1))



5. 유료 vs 무료 사용자 평균 나이 비교

In [21]:
paid_age = 0
paid_cnt = 0
free_age = 0
free_cnt = 0

for user in users:
  if user['is_paid'] == True:
    paid_age += user['age']
    paid_cnt += 1
  else:
    free_age += user['age']
    free_cnt += 1

paid_avg = paid_age / paid_cnt
free_avg = free_age / free_cnt

print("paid_avg", paid_avg)
print("free_avg", free_avg)

paid_avg 36.333333333333336
free_avg 27.0


"상황 설명"
=> 평균만 가지고, 데이터 전체를 확인하기 어렵기 때문에 추가적인 기술 통계량 필요

- 두 집단의 나이 평균에 차이가 있나요?
    - 전체 나이 32.6 세
    - 유료사용자의 평균 나이 36.3세

Hint! 구매력이 있는 / 구매 의사가 높은 그룹 = 나이 연관이 있을 것이다.  
➡️ 가설 설정에서 활용 가능